lib/utils/fields.js

```diff
@@ -89,12 +83,48 @@ export function dataObjectToArray(dataObject, fields, options = {}) {
     }
     const field = fields[fieldIdx];
     if (field.multivalued && Array.isArray(value)) {
-      dataArray[fieldIdx] = value
-        .map((v) => stringifyDatatype(v, field.datatype))
-        .join(multivalueDelimiter);
+      dataArray[fieldIdx] = formatMultivaluedValue(
+        value.map((v) => stringifyDatatype(v, field.datatype))
+      );
     } else {
       dataArray[fieldIdx] = stringifyDatatype(value, field.datatype);
     }
   }
   return dataArray;
 }
+
+/**
+ * Parse a formatted string representing a multivalued value and return an
+ * array of the individual values.
+ *
+ * @param {String} value String-formatted multivalued value.
+ * @return {Array<String>} Array of individual string values.
+ */
+export function parseMultivaluedValue(value) {
+  if (!value) {
+    return [];
+  }
+  // trim the delimiter and the resulting tokens to be flexible about what
+  // this function accepts
+  return value
+    .split(MULTIVALUED_DELIMITER.trim())
+    .map((v) => v.trim())
+    .filter((v) => !!v);
+}
+
+/**
+ * Format a string array of multivalued values into a single string representation.
+ *
+ * @param {Array<Any>} values Array of individual values.
+ * @return {String} String-formatted multivalued value.
+ */
+export function formatMultivaluedValue(values) {
+  if (!values) {
+    return '';
+  }
+
+  return values
+    .filter((v) => !!v)
+    .map((v) => (typeof v === 'string' ? v.trim() : String(v)))
```

Comment:
And this trims strings, and converts numeric to string?

Reply:
Correct. Technically anything that's not a string will get converted to a string, but yeah it's primary to handle numbers here.